app/views/unattended/provisioning_templates/snippet/subscription_manager_setup.erb

```diff
@@ -0,0 +1,128 @@
+<%#
+kind: snippet
+name: subscription_manager_setup
+model: ProvisioningTemplate
+snippet: true
+description: |
+  Streamline the provisioning process by configuring subscription-manager during the Global Registration template execution.
+  This snippet integrates into the existing "redhat_register" snippet for efficient host registration.
+-%>
+
+# Define the path to rhsm.conf
+RHSM_CFG=/etc/rhsm/rhsm.conf
+
+<% if plugin_present?('katello') -%>
+  # Define the path to the Katello server CA certificate
+  KATELLO_SERVER_CA_CERT=/etc/rhsm/ca/katello-server-ca.pem
+
+  # If SSL_CA_CERT is not set, create a temporary file for it
+  if [ -z "$SSL_CA_CERT" ]; then
+  SSL_CA_CERT=$(mktemp)
+  cat << EOF > "$SSL_CA_CERT"
+  <%= foreman_server_ca_cert %>
+  EOF
+  fi
+
+  <% if @existing_machine -%>
+    # Backup the original rhsm.conf file
+    if [ -f $RHSM_CFG ] ; then
+    test -f $RHSM_CFG.bak || cp $RHSM_CFG $RHSM_CFG.bak
+    fi
+
+    # rhn-client-tools conflicts with subscription-manager package
+    # since rhn tools replaces subscription-manager, we need to explicitly
+    # install subscription-manager after the rhn tools cleanup
+    if [ x$ID = xol ]; then
+    $PKG_MANAGER_REMOVE rhn-client-tools
+    $PKG_MANAGER_INSTALL --setopt=obsoletes=0 subscription-manager
+    fi
+
+    <% if truthy?(@force) -%>
+      # Unregister host and remove all local system and subscription data
+      if [ -x "$(command -v subscription-manager)" ] ; then
+      subscription-manager unregister || true
+      subscription-manager clean
+      fi
+
+      $PKG_MANAGER_REMOVE katello-ca-consumer\*
+    <% end -%>
+  <% end -%>
+
+  # Prepare the SSL certificate
+  mkdir -p /etc/rhsm/ca
+  cp -f $SSL_CA_CERT $KATELLO_SERVER_CA_CERT
+  chmod 644 $KATELLO_SERVER_CA_CERT
+
+<% end -%>
+
+# Prepare subscription-manager
+if ! [ -x "$(command -v subscription-manager)" ] ; then
+$PKG_MANAGER_INSTALL subscription-manager
+else
+$PKG_MANAGER_UPGRADE subscription-manager
+fi
+
+# Check if rhsm.conf exists
+if ! [ -f $RHSM_CFG ] ; then
+echo "'$RHSM_CFG' not found, cannot configure subscription-manager"
+<% if plugin_present?('katello') -%>
+  rm -f $SSL_CA_CERT
+<% end -%>
+exit 1
+fi
+
+<%
+  if plugin_present?('katello')
+    server_hostname = @rhsm_url&.host
+    server_port = @rhsm_url&.port
+    server_prefix = @rhsm_url&.path
+    repo_ca_cert = "$KATELLO_SERVER_CA_CERT"
+    rhsm_baseurl = @pulp_content_url
+  else
+    server_hostname = "subscription.rhsm.redhat.com"
+    server_port = "443"
+    server_prefix = "/subscription"
+    repo_ca_cert = "/etc/rhsm/ca/redhat-uep.pem"
+    rhsm_baseurl = "https://cdn.redhat.com"
```

Comment:
Just thinking out loud: is there a way to reset these to default values so we don't need to track these values?

Reply:
What do you mean by 'reset to default'? From what I know, the default is empty values. 
And where else will we use these vars, that we wouldn't need to track them?